### PR TITLE
theme and desktop changes

### PR DESCRIPTION
**Pull request by JoesCat**
Add Freedesktop desktop support files and also merge-in added
modifications from Tobias Mueller <muelli@cryptobitch.de>
Support starts from older v1.0 upto what appears currently in
/usr/share/{appdata,applications,icons,mime,metainfo,pixmaps}
Theme support also added plus additional images included for
pixmaps and 2012 themes. Renamed directories from icons-old and
icons to 2012 and tango (allowing for further theme expansion).

Closes issue #3372, #3056, #3053, #892, #3368 

@copyninja, @pnemade, @muelli - please check.

FYI, my internet is down, so I won't be able to reply quickly.
Add any patching you think necessary.
Thanks.

### CODE REVIEW

**Comment by frank-trampe** [comment]
I don't really understand what's happening, but I'm satisfied that this is safe to merge.

**Comment by frank-trampe** [comment]
Other stakeholders please comment promptly, as I would really like to tag a release before the end of the year.

**Comment by JoesCat** [comment]
Almost ok.... had too many arguments for AM_CONDITIONAL (4 is too many)....got a fix here by putting decision in an if/then decision, but internet problems are causing havoc.... hope to get replacement modem in a couple days.

  AM_CONDITIONAL to if/then
    AM_CONDITIONAL complained of too many arguments.

--------------------------------- configure.ac ---------------------------------
index ace319b..69549f8 100644
@@ -686,9 +686,13 @@ AC_PATH_PROG([DESKTOP_FILE_INSTALL],[desktop-file-install],[:])
 AC_PATH_PROG([UPDATE_MIME_DATABASE],[update-mime-database],[:])
 AC_PATH_PROG([UPDATE_DESKTOP_DATABASE],[update-desktop-database],[:])
 
-AM_CONDITIONAL([XDG_DESKTOP],
-   [test x"${fontforge_desktop}" != xno -a test x"${DESKTOP_FILE_INSTALL}" != "x:"
-    -a test x"${UPDATE_MIME_DATABASE}" != "x:" -a test x"${UPDATE_DESKTOP_DATABASE}" != "x:"])
+i_have_desktop=yes
+if test x"${fontforge_desktop}" = xno || test x"${DESKTOP_FILE_INSTALL}" = "x:"; then
+  i_have_desktop=no
+if test x"${UPDATE_MIME_DATABASE}" = "x:" || test x"${UPDATE_DESKTOP_DATABASE}" = "x:"; then
+  i_have_desktop=no
+fi
+AM_CONDITIONAL([XDG_DESKTOP],[test x"${i_have_desktopUPDATE_MIME_DATABASE}" = xyes])
 
 #--------------------------------------------------------------------------
 # Create and Set FontForge Compiler Flags

File: desktop/Makefile.am [hunk 1]
```diff
@@ -25,32 +25,107 @@
...
+if USE_THEME_TANGO
+ICON_SRC = $(srcdir)/tango
+endif
+
+if USE_THEME_2012
+ICON_SRC = $(srcdir)/2012
+endif
+
+desktopdir = $(PACKAGEROOTDIR)/applications
+desktop_DATA = fontforge.desktop
```
Comment (muelli) [inline desktop/Makefile.am]: Have you not changed the appid on purpose?
Otherwise these files, i.e. the desktop, appdata, and icon files, need to be renamed to resemble the appid, i.e. org.fontforge.FontForge.

File: desktop/fontforge.appdata.xml [hunk 2]
```diff
@@ -84,21 +83,8 @@
...
   <screenshots>
-    <screenshot type="default" height="675" width="1200">
-      <image>https://github.com/fontforge/fontforge/raw/master/doc/html/ff-screenshot.png</image>
-      <caption>Fontforge showing a glyph being edited</caption>
-    </screenshot>
+    <screenshot type="default" width="1200" height="675">https://github.com/fontforge/fontforge/raw/master/doc/html/ff-screenshot.png</screenshot>
   </screenshots>
   <url type="homepage">https://github.com/fontforge/fontforge</url>
   <update_contact>fontforge-devel@lists.sourceforge.net</update_contact>
-  <content_rating type="oars-1.1"/>
```
Comment (muelli) [inline desktop/fontforge.appdata.xml]: What's the purpose of removing OARS and release history and so on?

File: desktop/Makefile.am (hunk 1 above)
Comment (JoesCat) [reply desktop/Makefile.am]: I'll need to review this in better detail when I get propper internet again.
This patch attempts to try meet adding a desktop (requested), and also attempt to maintain older ver1 information, but also try and include newer data in the metainfo directory to meet latest version requirements.
Do you think making a duplicate of fontforge.metainfo.xml in /usr{/local}/share/appdata could be helpful? I did see a couple metainfo.xml in the appdata directory too.

**Comment by JoesCat** [comment]
@frank-trampe - This can be squash-merge when you're ready to go, but prefer to also see feedback.

if/when --enable-fontforge-desktop it will add desktop support files in /usr{/local}/share/{appdata,applications,pixmaps,icons,metainfo,mime,mime/packages}.
if/when --enable-fontforge-desktop=PATH, then it will apply files in location PATH instead.
If DESTDIR is empty, then assumption made that this is for user PC and not some other location.

Patch is based on looking at a whole bunch of different applications pushing files into the /usr/share area, plus what we need for FontForge, including tango, 2012, and potentially other themes.

**Comment by JoesCat** [comment]
@muelli - a quick check gives the impression we're looking at two different specs....
1 - desktop: https://specifications.freedesktop.org/desktop-entry-spec/1.2/ar01s02.html
2 - appstream: https://www.freedesktop.org/software/appstream/docs/
...the desktop spec (1) seems ok with maintaining "fontforge.desktop" in ver1.2
...I'll read-thru (2) since existing /usr directory examples I looked at didn't appear to show reverse dns naming except for (mainly) org.kde and one or two org.gnome files.

**Comment by muelli** [comment]
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic is relevant

**Comment by JoesCat** [comment]
Based on latest patches, is is best reduced from:
"Add ./configure --enable-fontforge-desktop"
to
"theme selection"

File: desktop/Makefile.am [hunk 3]
```diff
@@ -25,23 +25,109 @@
...
-EXTRA_DIST = icons/src/icon-16x16-apps-fontforge.svg	\
-	icons/src/icon-24x24-apps-fontforge.svg		\
-	icons/src/icon-32x32-apps-fontforge.svg
+PACKAGEROOTDIR = ${datarootdir}
+
+# Where .desktop files go.
+desktopdir = $(PACKAGEROOTDIR)/applications
+desktop_DATA = org.fontforge.FontForge.desktop
+
+appdatadir = $(PACKAGEROOTDIR)/appdata
```
Comment (jtanx) [inline desktop/Makefile.am]: What's going on here? In the old makefile, `org.fontforge.FontForge.appdata.xml` got installed to the metainfo folder, but now it gets installed to appdata instead? The AppImage packager expects the former.

File: desktop/Makefile.am [hunk 4]
```diff
@@ -25,23 +25,109 @@
...
+	org.fontforge.FontForge.appdata.xml		\
+	org.fontforge.FontForge.desktop			\
+	fontforge.xml					\
+	org.fontforge.FontForge.metainfo.xml
+
+install-data-hook: touch-icon-dir
+uninstall-hook: touch-icon-dir
+
+touch-icon-dir:
+	if test -z "$(DESTDIR)"; then $(TOUCH) $(DESTDIR)$(HICOLORDIR) || : ; fi
```
Comment (jtanx) [inline desktop/Makefile.am]: Why is this needed?

File: desktop/Makefile.am (hunk 3 above)
Comment (muelli) [reply desktop/Makefile.am]: FTR: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html mentions `appdata` being legacy and `metainfo` being the directory du jour.
This may be related to [this change](https://github.com/fontforge/fontforge/pull/3387#discussion_r244451928) I haven't understood.

Comment (JoesCat) [reply desktop/Makefile.am]: On older distro installs, everything is in appdata, should continue to keep appdata in there.
I'll clone another copy into metainfo too.
I've looked at a couple of recent distro /usr/share/metainfo and it's sort of a jumble of both types (bit of a mess in there).

Also removing the 'Extra_DIST = desktop' I added in the main Makefile.am as this isn't the right way to 'make dist' anyways.

File: desktop/Makefile.am (hunk 4 above)
Comment (JoesCat) [reply desktop/Makefile.am]: found that tidbit of info while looking at all sorts of desktop installs:
https://github.com/GNOME/gimp/blob/master/desktop/Makefile.am
It was ADDDESKTOP earlier. This seemed a bit more descriptive.
take it out?

File: desktop/Makefile.am (hunk 3 above)
Comment (JoesCat) [reply desktop/Makefile.am]: @muelli - directory du jour is a good description. Some specs seems to shift with time :-)
To avoid having to re-install, you may want to look at a couple of live distros inside those directories, I'd suggest a couple that have been around for quite a while.... look at knoppix 6, puppy linux, then compare it with a couple of recent ones, like mageia 7, ubuntu, etc. This way you avoid installing unnecessary programs but at the same time get a lay-of-the-land.

File: desktop/Makefile.am (hunk 4 above)
Comment (jtanx) [reply desktop/Makefile.am]: Yeah let's keep it simple and not do this
